ui/apps/everest/src/pages/database-form/database-form.tsx

```diff
@@ -176,157 +139,54 @@ export const DatabasePage = () => {
-              {activeStep === steps.length - 1 ? (
-                <Button
-                  onClick={handleSubmit(onSubmit)}
-                  variant="contained"
-                  disabled={isCreating || isUpdating}
-                  data-testid="db-wizard-submit-button"
-                >
-                  {mode === 'edit'
-                    ? Messages.editDatabase
-                    : Messages.createDatabase}
-                </Button>
-              ) : (
-                <Button
-                  onClick={handleNext}
-                  variant="contained"
-                  data-testid="db-wizard-continue-button"
-                  disabled={Object.values(errors).length > 0}
-                >
-                  {Messages.continue}
-                </Button>
-              )}
-            </Box>
-          </form>
-          {isDesktop ? (
-            <Drawer
-              variant="permanent"
-              anchor="right"
-              sx={{
-                // MuiDrawer-paper will take 25% of the whole screen because it has a "fixed" positioning
-                // Hence, we must use vw here to have the same calculation
-                // We subtract the padding
-                width: (theme) => `calc(25vw - ${theme.spacing(4)})`,
-                flexShrink: 0,
-                ml: 3,
-                [`& .MuiDrawer-paper`]: {
-                  width: '25%',
-                  boxSizing: 'border-box',
-                },
-              }}
-            >
-              <Toolbar />
-              {PreviewContent}
-            </Drawer>
-          ) : (
-            <>
-              <Divider
-                orientation="horizontal"
-                flexItem
-                sx={{
-                  // This is a little tweak
-                  // We make the divider longer, adding the main padding value
-                  // Then, to make it begin before the main padding, we add a negative margin
-                  // This way, the divider will cross the whole section
-                  width: `calc(100% + ${theme.spacing(4 * 2)})`,
-                  ml: -4,
-                  mt: 6,
-                }}
-              />
-              {PreviewContent}
-            </>
-          )}
-        </Stack>
-      </FormProvider>
+      <Stack direction={isDesktop ? 'row' : 'column'}>
+        <FormProvider {...methods}>
+          <DatabaseFormBody
+            activeStep={activeStep}
+            longestAchievedStep={longestAchievedStep}
+            disableNext={formHasErrors}
+            disableSubmit={formHasErrors || isCreating || isUpdating}
+            onSubmit={handleSubmit(onSubmit)}
+            onCancel={() => setCancelModalOpen(true)}
+            handleNextStep={handleNext}
+            handlePreviousStep={handleBack}
+          />
+          <DatabaseFormSideDrawer
+            activeStep={activeStep}
+            longestAchievedStep={longestAchievedStep}
+            handleSectionEdit={handleSectionEdit}
```

Comment:
I'm just thinking... Why we clear Errors during handle Section Edit? How to handle situation when user do error in part of the form... i think we should let him know about it. Maybe... a red marker near the preview section? Because now user can did an error and then just jump to the last part of form via edit and in the best case he will get an error, in worth - nothing informal why the button is not available

this is not a question for this PR, but if we do refactoring, this is one of the pressing problems when the user doesn't work with the form step by step

Reply:
That's a fair point. I think what happens is that we clear the errors because we don't have a way to then tell users some step behind has an error. So yeah, maybe a red icon could be useful. I'll bring this topic to the Triad